Publish Magento Cloud 2FA for SSH beta documentation (#7517)
magento/devdocs: src/cloud/project/project-enable-mfa-project.md

---
group: cloud-guide
title: Enable MFA enforcement for SSH access
functional_areas:
  - Cloud
  - Configuration
---

> For added security, {{site.data.var.ece}} provides an mutlitfactor authentication (MFA) enforcement feature to manage multifactor authentication requirements for SSH access to Cloud infrastructure.

Author: ```suggestion
For added security, {{site.data.var.ece}} provides a mutlitfactor authentication (MFA) enforcement feature to manage multifactor authentication requirements for SSH access to Cloud infrastructure.
```

> When this feature is enabled on a project, any user or machine account establishing an SSH connection to an environment must follow an authentication workflow that requires a two-factor authentication (2FA) code and an SSH certificate, or an API token to access the environment.

Reviewer: ```suggestion
When MFA enforcement is enabled on a project, any user or machine account establishing an SSH connection to an environment must follow an authentication workflow that requires a two-factor authentication (2FA) code and an SSH certificate, or an API token to access the environment.
```

> {:bs.callout-important}

Author: ```suggestion
{:.bs.callout-important}
```

The MFA enforcement feature is not enabled by default on {{site.data.var.ece}} projects. You must submit a Magento support ticket to request 2FA enablement on your project. After enablement, all subsequent SSH connection requests require either a 2FA token or an API token.

## Certificates for SSH access

The MFA enforcement feature allows users to exchange an OAUTH access token with a short-lived SSH certificate generated by the Magento Cloud Certifier API.

> If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.

Reviewer: Admin is in caps, but not contributor.

> If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.

Author: ```suggestion
If the user accessing the environment has project Admin or Contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```

> If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code or API token, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.

Author: ```suggestion
If the user accessing the environment has project Admin or contributor access, a valid SSH key, and a valid 2FA code, {{site.data.var.ece}} uses these credentials to generate the temporary SSH certificate. The certificate expiration is set to 1 hour, but it refreshes automatically during the current session.
```

> After logging into a project with MFA enforcement, users must run the following command to generate the certificate:

Reviewer: ```suggestion
After logging into a project with MFA enforcement, users must use the CLI to generate the SSH certificate:
```

```bash
magento-cloud ssh-cert:load
```

> This command generates the SSH certificate and installs it in the SSH agent of the local user.

Reviewer: ```suggestion
The `ssh-cert:load` command generates the SSH certificate and installs it in the SSH agent of the local user.
```

> This command generates the SSH certificate and installs the certificate in the SSH agent of the local user.

Author: ```suggestion
This command generates the SSH certificate and installs it in the SSH agent of the local user.
```

## Connect to an environment using SSH with 2FA

When MFA enforcement is enabled on a {{site.data.var.ece}} project, any user that connects to a Cloud environment using SSH must have 2FA enabled on their account. See [Enable 2FA][].

**Prerequisites:**

For {{site.data.var.ece}} projects enabled for MFA enforcement, SSH access requires the following permissions and account settings:

-  [Admin or contributor access to the {{site.data.var.ece}} environment][Manage user access]
-  [SSH access key configured on account][add public SSH key]
-  [2FA enabled on account][]

{:.procedure}
To connect using SSH with 2FA user account credentials:

1. Log in to your {{site.data.var.ece}} account and authenticate using 2FA.

> 1. After logging in, open a command line and run the following command to generate the certificate.

Reviewer: open a command line?
run the command is used too much.

```suggestion
1. On your local workstation, use the CLI to generate an SSH certificate.
```

   ```bash
   magento-cloud ssh-cert:load
   ```

   ```terminal
   Generating SSH certificate...
     Expires at: 2020-07-13T15:28:13-04:00
     Multi-factor authentication: verified
     Mode: interactive
   The certificate will be automatically refreshed when necessary.
   Checking SSH configuration file: /Users/<user-name>/.ssh/config
   Do you want to update the file automatically? [Y/n] Y
   Configuration file updated successfully: /Users/<user-name>/.ssh/config
   ```

> 1. Connect to the project using SSH:

Reviewer: Must this be master? Probably need to be specific.

> 1. Connect to the project using SSH:

Author: ```suggestion
1. Connect to the Cloud environment using SSH:
```
